network/src/protocols/health_checker/test.rs

```diff
@@ -15,136 +15,148 @@ use crate::{
+            ping_failures_tolerated,
+        );
+
+        (
+            Self {
+                mock_time: mock_time.into_mock(),
+                peer_mgr_reqs_rx,
+                peer_mgr_notifs_tx,
+                connection_reqs_rx,
+                connection_notifs_tx,
+            },
+            health_checker,
+        )
+    }
 
-async fn expect_ping(
-    network_reqs_rx: &mut diem_channel::Receiver<(PeerId, ProtocolId), PeerManagerRequest>,
-) -> (Ping, oneshot::Sender<Result<Bytes, RpcError>>) {
-    let req = network_reqs_rx.next().await.unwrap();
-    let rpc_req = match req {
-        PeerManagerRequest::SendRpc(_peer_id, rpc_req) => rpc_req,
-        _ => panic!("Unexpected PeerManagerRequest: {:?}", req),
-    };
+    fn new_strict() -> (Self, HealthChecker) {
+        Self::new_permissive(0 /* ping_failures_tolerated */)
+    }
+
+    async fn trigger_ping(&self) {
+        self.mock_time.advance_async(PING_INTERVAL).await;
+    }
 
-    let protocol_id = rpc_req.protocol_id;
-    let req_data = rpc_req.data;
-    let res_tx = rpc_req.res_tx;
+    async fn expect_ping(&mut self) -> (Ping, oneshot::Sender<Result<Bytes, RpcError>>) {
+        let req = self.peer_mgr_reqs_rx.next().await.unwrap();
+        let rpc_req = match req {
+            PeerManagerRequest::SendRpc(_peer_id, rpc_req) => rpc_req,
+            _ => panic!("Unexpected PeerManagerRequest: {:?}", req),
+        };
 
-    assert_eq!(protocol_id, ProtocolId::HealthCheckerRpc,);
+        let protocol_id = rpc_req.protocol_id;
+        let req_data = rpc_req.data;
+        let res_tx = rpc_req.res_tx;
 
-    match bcs::from_bytes(&req_data).unwrap() {
-        HealthCheckerMsg::Ping(ping) => (ping, res_tx),
-        msg => panic!("Unexpected HealthCheckerMsg: {:?}", msg),
+        assert_eq!(protocol_id, ProtocolId::HealthCheckerRpc,);
+
+        match bcs::from_bytes(&req_data).unwrap() {
+            HealthCheckerMsg::Ping(ping) => (ping, res_tx),
+            msg => panic!("Unexpected HealthCheckerMsg: {:?}", msg),
+        }
     }
-}
 
-async fn expect_ping_send_ok(
-    network_reqs_rx: &mut diem_channel::Receiver<(PeerId, ProtocolId), PeerManagerRequest>,
-) {
-    let (ping, res_tx) = expect_ping(network_reqs_rx).await;
-    let res_data = bcs::to_bytes(&HealthCheckerMsg::Pong(Pong(ping.0))).unwrap();
-    res_tx.send(Ok(res_data.into())).unwrap();
-}
+    async fn expect_ping_send_ok(&mut self) {
+        let (ping, res_tx) = self.expect_ping().await;
+        let res_data = bcs::to_bytes(&HealthCheckerMsg::Pong(Pong(ping.0))).unwrap();
+        res_tx.send(Ok(res_data.into())).unwrap();
+    }
 
-async fn expect_ping_send_notok(
-    network_reqs_rx: &mut diem_channel::Receiver<(PeerId, ProtocolId), PeerManagerRequest>,
-) {
-    let (_ping_msg, res_tx) = expect_ping(network_reqs_rx).await;
-    // This mock ping request must fail.
-    res_tx.send(Err(RpcError::TimedOut)).unwrap();
-}
+    async fn expect_ping_send_notok(&mut self) {
```

Comment:
Nit: Put an `_` between `not` and `ok`.  I read over this a few times, and read `no tok`.